ProfileService/ProfileService/Controllers/ProfileController.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using ProfileService.BLL.Dto;
+using ProfileService.BLL.Services.IServices;
+
+namespace ProfileService.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class ProfileController(IUserProfileService profileService) : ControllerBase
+    {
+        [AllowAnonymous]
+        [HttpPost]
+        public async Task<IActionResult> CreateProfile(UserRegistrationData userData, CancellationToken cancellationToken)
```

Comment:
No need in IActionResult. Controller will return success status codes anyway and ExceptionMiddleware will handle all client errors